mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/InvoiceEventsDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+import static org.folio.util.AuditEventDBConstants.ACTION_FIELD;
+import static org.folio.util.AuditEventDBConstants.EVENT_DATE_FIELD;
+import static org.folio.util.AuditEventDBConstants.ID_FIELD;
+import static org.folio.util.AuditEventDBConstants.INVOICE_ID_FIELD;
+import static org.folio.util.AuditEventDBConstants.MODIFIED_CONTENT_FIELD;
+import static org.folio.util.AuditEventDBConstants.ORDER_BY_PATTERN;
+import static org.folio.util.AuditEventDBConstants.TOTAL_RECORDS_FIELD;
+import static org.folio.util.AuditEventDBConstants.USER_ID_FIELD;
+import static org.folio.util.DbUtils.formatDBTableName;
+
+@Repository
+public class InvoiceEventsDaoImpl implements InvoiceEventsDao {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  public static final String TABLE_NAME = "acquisition_invoice_log";
+
+  public static final String GET_BY_INVOICE_ID_SQL = "SELECT id, action, invoice_id, user_id, event_date, action_date, modified_content_snapshot," +
+    " (SELECT count(*) AS total_records FROM %s WHERE invoice_id = $1) FROM %s WHERE invoice_id = $1 %s LIMIT $2 OFFSET $3";
+
+  public static final String INSERT_SQL = "INSERT INTO %s (id, action, invoice_id, user_id, event_date, action_date, modified_content_snapshot)" +
+    " VALUES ($1, $2, $3, $4, $5, $6, $7)";
+
+  private final PostgresClientFactory pgClientFactory;
+
+  public InvoiceEventsDaoImpl(PostgresClientFactory pgClientFactory) {
+    this.pgClientFactory = pgClientFactory;
+  }
+
+  @Override
+  public Future<RowSet<Row>> save(InvoiceAuditEvent invoiceAuditEvent, String tenantId) {
+    LOGGER.debug("save:: Saving Invoice AuditEvent with tenant id : {}", tenantId);
+    Promise<RowSet<Row>> promise = Promise.promise();
+    LOGGER.debug("formatDBTableName:: Formatting DB Table Name with tenant id : {}", tenantId);
+    String logTable = formatDBTableName(tenantId, TABLE_NAME);
+    String query = format(INSERT_SQL, logTable);
+    makeSaveCall(promise, query, invoiceAuditEvent, tenantId);
+    LOGGER.info("save:: Saved Invoice AuditEvent with tenant id : {}", tenantId);
+    return promise.future();
+  }
+
+  @Override
+  public Future<InvoiceAuditEventCollection> getAuditEventsByInvoiceId(String invoiceId, String sortBy, String sortInvoice, int limit, int offset, String tenantId) {
+    LOGGER.debug("getAuditEventsByInvoiceId:: Retrieving AuditEvent with invoice id : {}", invoiceId);
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      LOGGER.debug("formatDBTableName:: Formatting DB Table Name with tenant id : {}", tenantId);
+      String logTable = formatDBTableName(tenantId, TABLE_NAME);
+      String query = format(GET_BY_INVOICE_ID_SQL, logTable, logTable,  format(ORDER_BY_PATTERN, sortBy, sortInvoice));
+      Tuple queryParams = Tuple.of(UUID.fromString(invoiceId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
```

Comment:
catch is not needed here, in case of failures we will have failed future.
Potentailly it can be a chance of runtime excpeptions even before creating futures, but it is very low I think. And if there is for example some DB failure - this catch will not help and if you want to log - .onFailure can be used